Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -196,6 +196,48 @@ class MessageDTO: NSManagedObject {
         request.fetchOffset = offset
         return try! context.fetch(request)
     }
+
+    static func loadAttachmentCounts(
+        for messageId: MessageId,
+        context: NSManagedObjectContext
+    ) -> [AttachmentType: Int] {
+        enum AttachmentScoreKey: String {
+            case type
+            case count
+        }
+
+        let count = NSExpressionDescription()
+        count.name = AttachmentScoreKey.count.rawValue
+        count.expressionResultType = .integer64AttributeType
+        count.expression = NSExpression(
+            forFunction: "count:",
+            arguments: [NSExpression(forKeyPath: AttachmentScoreKey.type.rawValue)]
+        )
+
+        let request = NSFetchRequest<NSDictionary>(entityName: AttachmentDTO.entityName)
+        request.propertiesToFetch = [AttachmentScoreKey.type.rawValue, count]
+        request.propertiesToGroupBy = [AttachmentScoreKey.type.rawValue]
+        request.resultType = .dictionaryResultType
+        request.predicate = NSPredicate(
+            format: "%K.%K == %@",
+            #keyPath(AttachmentDTO.message),
+            #keyPath(MessageDTO.id),
+            messageId
+        )
+
+        guard let results = try? context.fetch(request) else {
+            log.error("Failed to fetch attachment counts for the message with id: \(messageId)")
+            return [:]
+        }
```

Comment:
Maybe it's better to capture the error and put it into `log.error`?

Reply:
✅ Done